src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -132,7 +132,14 @@ public ActionEndpointFactory(RoutePatternTransformer routePatternTransformer)
                 var updatedRoutePattern = _routePatternTransformer.SubstituteRequiredValues(resolvedRoutePattern, resolvedRouteValues);
                 if (updatedRoutePattern == null)
                 {
-                    throw new InvalidOperationException("Failed to update route pattern with required values.");
```

Comment:
@JamesNK - I remember when we worked on this we really weren't sure what a user could do to trigger this error, but we wanted to make it blow up just for insurance.

Well, someone found a cause that causes it to fail. 👍